### PR TITLE
Add a debug logger around rendition generation

### PR DESCRIPTION
**Pull request by RealOrangeOne**
This allows insight into which images are taking the longest to generate, which fail to render at all, and potentially which images are causing crashes (as in they start, but never stop).

The logging is intentionally only on DEBUG level, so it's opt-in, and is also reasonably quiet so it doesn't bloat logs.

Relevant Slack discussion: https://wagtailcms.slack.com/archives/C023WG6FB7C/p1627922996003600.

### CODE REVIEW

File: wagtail/images/models.py [hunk 1]
```diff
@@ -303,7 +308,23 @@ def get_rendition(self, filter):
             )
         except Rendition.DoesNotExist:
             # Generate the rendition image
-            generated_image = filter.run(self, BytesIO())
+            try:
+                LOGGER.debug("Generating '%s' rendition for image %d", (
```
Comment (RealOrangeOne) [inline wagtail/images/models.py:312]: I'm not entirely sure if this log before is necessary.

My main intention is to cover the case where generating an image uses so much memory that the application either locks up or is OOM killed (or platform-specific kill). I suspect an exception handler probably wouldn't catch that.

Question is, is it important enough to have 2 log messages for each generation as opposed to just 1?

File: wagtail/images/models.py [hunk 2]
```diff
@@ -29,6 +31,9 @@
 from wagtail.search.queryset import SearchableQuerySetMixin
 
 
+LOGGER = logging.getLogger("wagtail.images")
```
Comment (RealOrangeOne) [inline wagtail/images/models.py:34]: Question: Should we add some explicit documentation about the existence of this logger, and how to enable / use it?

**Comment by gasman** [comment]
Looks good to me... Merged in 90c86e76d0d6f46208eb1efb2918026d61ee3326.

**Comment by jgeskens** [comment]
Hi, this patch seems to break the generation of image renditions. This logger statement uses incorrect syntax for the format arguments (a tuple instead of *args), triggering an exception: https://github.com/wagtail/wagtail/commit/90c86e76d0d6f46208eb1efb2918026d61ee3326#diff-c195bc8bd4d5a11ae02d0451edb4dd4a8e7b0014398a384a1be3ff3c355a5ee8R312
Before this change, everything still worked fine. So for now I have to downgrade.

**Comment by lb-** [comment]
@jgeskens thanks for the comment, are you able to raise a bug for this, it will help us action this sooner, comments against closed PRs are easy to get lost.

**Comment by RealOrangeOne** [comment]
Nice catch - i'm not sure how I missed that, or how CI never caught it. I've opened a fix as #7694

**Comment by lb-** [comment]
All good - fix PR is already up https://github.com/wagtail/wagtail/pull/7694

**Pull request closed without merging**